File: app/src/main/java/com/ptithcm/thuan6420/basecleanarchitecture/data/datasources/sharepreferences/UserSharedPreferences.java
package com.ptithcm.thuan6420.basecleanarchitecture.data.datasources.sharepreferences;

import android.content.Context;
import android.content.SharedPreferences;

public class UserSharedPreferences {
    private static final String USER_SHARED_PREFERENCES = "USER_SHARED_PREFERENCES";
    private Context mContext;

    public UserSharedPreferences(Context mContext) {
        this.mContext = mContext;
    }

    public void putUserValue(String key, String value) {
        SharedPreferences sharedPreferences = mContext.getSharedPreferences(USER_SHARED_PREFERENCES,
                Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public String getUserValue(String key) {
        SharedPreferences sharedPreferences = mContext.getSharedPreferences(USER_SHARED_PREFERENCES,
                Context.MODE_PRIVATE);
        return sharedPreferences.getString(key, "");
    }
}

File: app/src/main/java/com/ptithcm/thuan6420/basecleanarchitecture/data/datasources/sharepreferences/UserSharedPreferences.kt
package com.ptithcm.thuan6420.basecleanarchitecture.data.datasources.sharepreferences

import com.google.gson.Gson
import com.ptithcm.thuan6420.basecleanarchitecture.ui.login.User
import com.ptithcm.thuan6420.basecleanarchitecture.Constants.PREF_REGISTERED_USER

class UserSharedPreferences() {

    fun getUserFromLocal(): User {
        val jsonUser = AppSharedPreferences.getSharedPreferencesValue(PREF_REGISTERED_USER)
        return Gson().fromJson(jsonUser, User::class.java)
    }

    fun setUserFromLocal(user: User) {
        val jsonUser = Gson().toJson(user)
        AppSharedPreferences.putSharedPreferencesValue(PREF_REGISTERED_USER, jsonUser)
    }

    companion object {
        private var Instance: UserSharedPreferences? = null
        operator fun invoke() = synchronized(this) {
            if (Instance != null) {
                Instance
            }
            Instance = UserSharedPreferences()
        }
    }
}